src/coprocessor/endpoint.rs

```diff
@@ -106,7 +106,9 @@ impl<E: Engine> Endpoint<E> {
 
     fn check_memory_locks(&self, req_ctx: &ReqContext) -> Result<()> {
         let start_ts = req_ctx.txn_start_ts;
-        self.concurrency_manager.update_max_ts(start_ts);
+        if !req_ctx.context.get_stale_read() {
```

Comment:
Need stale-read check memory locks? If `resolved-ts` reaches the `stale-read-ts`, there is no lock before the ts.

Reply:
There may have locks. For example,  right after the `resolved-ts` is advanced with a ts from PD, a prewrite request with a smaller `start_ts` come, in this case, `resolved-ts` won't decrease. 

Although `resolved-ts` guarantee that it must larger than any incoming `commit_ts`, so it is okay to ignore locks that less than `resolved-ts`. But I think it is too risky and not much benefit since reading historical data will encounter less locks, so currently, stale read request will check both the memory locks and persisted locks.